src/app/api/quote/route.ts

```diff
@@ -71,50 +46,48 @@ export async function POST(req: NextRequest) {
       return BadRequest('Bad Request')
     }
 
-    const rpcUrl = getAlchemyBaseUrl(chainId) + process.env.ALCHEMY_API_KEY
-    const zeroexSwapQuoteProvider =
-      getConfiguredZeroExSwapQuoteProvider(chainId)
-    const quoteProvider = new FlashMintQuoteProvider(
-      rpcUrl,
-      zeroexSwapQuoteProvider,
-    )
-
     const isMinting = outputToken.isIndex
-    const quoteRequest: FlashMintQuoteRequest = {
-      isMinting,
-      inputToken: inputToken.quoteToken,
-      outputToken: outputToken.quoteToken,
-      indexTokenAmount: BigNumber.from(
-        isMinting ? outputAmount! : inputAmount!,
-      ),
-      slippage,
+    const quoteRequest: {
+      account: string
+      chainId: string
+      inputToken: string
+      outputToken: string
+      inputAmount?: string
+      outputAmount?: string
+      slippage: string
+    } = {
+      account,
+      chainId: String(chainId),
+      inputToken: inputToken.quoteToken.address,
+      outputToken: outputToken.quoteToken.address,
+      slippage: String(slippage),
     }
-    if (isMintingIcUsd || isRedeemingIcUsd) {
-      quoteRequest.inputTokenAmount =
-        BigNumber.from(inputAmount) ?? BigNumber.from(0)
+    if (isMinting) {
+      quoteRequest.outputAmount = outputAmount
+    } else {
+      quoteRequest.inputAmount = inputAmount
     }
+    if (isMintingIcUsd) {
+      quoteRequest.inputAmount = inputAmount ?? '0'
+    }
+
+    const query = new URLSearchParams(quoteRequest).toString()
+    const url = `https://api.indexcoop.com/v2/quote?${query}`
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'x-api-key': process.env.INDEX_COOP_API_V2_KEY,
```

Comment:
Not added in defaults as it will be added with https://github.com/IndexCoop/index-app/pull/1617/files#diff-911a7630b911362b3d36403b554b3cc69f808fb1a44ffce5917a5ae853c45532R11